processing/src/main/java/org/apache/carbondata/processing/newflow/steps/SortProcessorStepImpl.java

```diff
@@ -74,7 +81,6 @@ public void initialize() throws IOException {
   public Iterator<CarbonRowBatch>[] execute() throws CarbonDataLoadingException {
     final Iterator<CarbonRowBatch>[] iterators = child.execute();
     Iterator<CarbonRowBatch>[] sortedIterators = sorter.sort(iterators);
-    child.close();
```

Comment:
It is removed as part of refactoring of code to support batch sort. Earlier it was needed to get the cardinality of all dimensions. Now there is a new interface `DictionaryCardinalityFinder` has been added to figure out the cardinalities.